frontend/src/app/features/auth/sign-in/sign-in.component.html

```diff
@@ -0,0 +1,31 @@
+<form class="sign-in-wrapper" [formGroup]="signInGroup">
+    <h1 i18n>Log into your account</h1>
+    <mat-form-field class="sign-in-field">
+        <mat-label i18n>Email</mat-label>
+        <input formControlName="email" matInput type="text" (keydown)="updateLoginErrors()"/>
+        <mat-error i18n *ngIf="signInGroup.get('email')?.hasError('email')">
+            Invalid email address
+        </mat-error>
+        <mat-error i18n *ngIf="signInGroup.get('email')?.hasError('invalidLogin')">
+            Either email or password is invalid, try again
+        </mat-error>
+    </mat-form-field>
+    <mat-form-field class="sign-in-field">
+        <mat-label i18n>Password</mat-label>
+        <input formControlName="password" matInput type="password" (keydown)="updateLoginErrors()"/>
+        <mat-error i18n *ngIf="!signInGroup.get('password')?.hasError('pattern') &&
```

Comment:
such complex conditions are hardly readable. you move it to function with clear name and use it. but in general it is a bad practive to use functions for bindings everywhere - it may affect perfomance: https://medium.com/showpad-engineering/why-you-should-never-use-function-calls-in-angular-template-expressions-e1a50f9c0496

Reply:
Simplified expressions using function for checking for length errors. In the future I will look for ways to avoid function binding when creating components